### PR TITLE
api refactoring with bolt

### PR DESCRIPTION
**Pull request by mekan-allaberdi**

### CODE REVIEW

File: src/slack/slack.js [hunk 1]
```diff
@@ -239,9 +240,8 @@ export class Slack {
   showError(channel, ts, msg) {
     const text = `:exclamation: ${msg}`
     logger.warn(`showed error to user: ${text}`)
-    return this.apiCall(ts ? 'chat.update' : 'chat.postMessage', {
-      channel, ts, as_user: true, text, attachments: [],
-    })
+
+    return ts ? this.boltApp.client.chat.update({channel, ts, as_user: true, text, attachments: []}) : this.boltApp.client.chat.postMessage(params)
```
Comment (rtrembecky) [inline src/slack/slack.js:244]: I think the `params` here are a mistake. and you could do something like the previous code structure, check this:
```js
return this.boltApp.client.chat[ts ? 'update' : 'postMessage']({channel, ts, as_user: true, text, attachments: []})
```

File: src/slack/slack.js [hunk 2]
```diff
@@ -580,9 +582,10 @@ export class Slack {
     const MESSAGES = VARIANT_MESSAGES[this.variant]
 
     const updateMessage = async (attachmentUpdate) => {
-      await this.apiCall('chat.update', {channel, ts, as_user: true,
-        attachments: [{...attachment, ...attachmentUpdate}],
-      })
+    //   await this.apiCall('chat.update', {channel, ts, as_user: true,
+    //     attachments: [{...attachment, ...attachmentUpdate}],
+    //   })
```
Comment (rtrembecky) [inline src/slack/slack.js:587]: old commented code

File: src/slack/slack.js [hunk 3]
```diff
@@ -605,7 +608,7 @@ export class Slack {
             fields: this.getOrderFields(order),
           })
 
-          await this.apiCall('chat.postMessage', {
+          this.boltApp.client.chat.postMessage({
```
Comment (rtrembecky) [inline src/slack/slack.js:611]: I see the await here is missing, is it intentional?

File: src/slack/slack.js [hunk 4]
```diff
@@ -618,7 +621,7 @@ export class Slack {
             completeNewMsg = true
           } else if (event.type === 'action' && event.actions[0].name === 'cancel') {
             await cancelOrder()
-            await this.apiCall('chat.postMessage', {
+            this.boltApp.client.chat.postMessage({
```
Comment (rtrembecky) [inline src/slack/slack.js:624]: missing await also here, intentional or not?

File: src/slack/slack.js [hunk 5]
```diff
@@ -821,7 +824,7 @@ export class Slack {
 
 
   async addReaction(name, channel, timestamp) {
-    const {ok, error} = await this.apiCall('reactions.add', {name, channel, timestamp}, {passError: true})
+    const {ok, error} = await this.boltApp.client.reactions.add({name, channel, timestamp}, {passError: true})
```
Comment (rtrembecky) [inline src/slack/slack.js:827]: do you know if `passError` here works?

Comment (mekan-allaberdi) [reply src/slack/slack.js:827]: not sure, removed it.

File: src/slack/slack.js (hunk 1 above)
Comment (mekan-allaberdi) [reply src/slack/slack.js:244]: I think here I wanted to assign `const params = {channel, ts, as_user: true, text, attachments: []}` and put params in both places.
Changed as you suggested.

File: src/slack/slack.js (hunk 2 above)
Comment (mekan-allaberdi) [reply src/slack/slack.js:587]: removed

File: src/slack/slack.js (hunk 5 above)
Comment (rtrembecky) [reply src/slack/slack.js]: yeah... it shouldn't work, it is an argument to our own `apiCall` function

**Review by rtrembecky — APPROVED** [review]
looks good. it is kind of scary, because some error handling will probably be broken, but you will basically look into it in the next task

**Pull request merged**